proxy/src/connection.rs

```diff
@@ -38,7 +38,17 @@ pub enum Connection {
 
 /// A trait describing that a type can peek (such as MSG_PEEK).
 pub trait Peek {
-    fn peek(&mut self, buf: &mut [u8]) -> io::Result<usize>;
+
+    fn peek(&mut self, buf: &mut [u8]) -> io::Result<usize> {
```

Comment:
What is the reason for needing to go from `Poll` -> `WouldBlock`? Is it to change less code?

Reply:
Previously, `tokio_io` provided a `peek()` function that returned `WouldBlock`, and we implemented a future around that. Now, `peek()` is deprecated in favour of `poll_peek()`, but we have code that depends on both synchronous and asynchronous peek functions.

I originally just removed the synchronous `peek` in favour of `poll_peek` but it broke code elsewhere.